weld/subsystem/src/main/java/org/jboss/as/weld/deployment/processors/ExternalBeanArchiveProcessor.java

```diff
@@ -205,24 +229,121 @@ public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitPro
         }
     }
 
-    private Set<URL> findExportedLocalBeansXml(Module dependencyModule) {
+    /**
+     *
+     * @param beanDiscoveryMode
+     * @param beansXmlUrl
+     * @param indexUrl
+     * @param beanDefiningAnnotations
+     * @return the set of discovered bean classes or null if unable to handle the provided beans.xml url
+     */
+    private Collection<String> discover(BeanDiscoveryMode beanDiscoveryMode, URL beansXmlUrl, URL indexUrl, Set<AnnotationType> beanDefiningAnnotations) {
+        List<String> discoveredClasses = new ArrayList<String>();
+        BiConsumer<String, ClassFile> consumer;
+
+        if (BeanDiscoveryMode.ANNOTATED.equals(beanDiscoveryMode)) {
+            // We must only consider types with bean defining annotations
+            Index index = tryLoadIndex(indexUrl);
+            if (index != null) {
+                // Use the provided index to find ClassInfo
+                consumer = (name, classFile) -> {
+                    ClassInfo classInfo = index.getClassByName(DotName.createSimple(name));
+                    if (classInfo != null && hasBeanDefiningAnnotation(classInfo, beanDefiningAnnotations)) {
+                        discoveredClasses.add(name);
+                    }
+                };
+            } else {
+                // Build ClassInfo on the fly
+                Indexer indexer = new Indexer();
+                consumer = (name, classFile) -> {
+                    try {
+                        ClassInfo classInfo = indexer.index(classFile.openStream());
```

Comment:
It does not look like this stream is closed?

Reply:
You're right, I'll fix it right away. Thanks!